xrootd/origin_test.go

```diff
@@ -0,0 +1,104 @@
+//go:build !windows
+
+/***************************************************************
+ *
+ * Copyright (C) 2023, Pelican Project, Morgridge Institute for Research
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ***************************************************************/
+
+package xrootd
+
+import (
+	"context"
+	"crypto/elliptic"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/pelicanplatform/pelican/config"
+	"github.com/pelicanplatform/pelican/daemon"
+	"github.com/pelicanplatform/pelican/origin_ui"
+	"github.com/pelicanplatform/pelican/param"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/require"
+)
+
+func TestOrigin(t *testing.T) {
+	viper.Reset()
+
+	viper.Set("Origin.ExportVolume", t.TempDir()+":/test")
+	// Disable functionality we're not using (and is difficult to make work on Mac)
+	viper.Set("Origin.EnableCmsd", false)
+	viper.Set("Origin.EnableMacaroons", false)
+	viper.Set("Origin.EnableVoms", false)
+	viper.Set("TLSSkipVerify", true)
+
+	viper.Set("ConfigDir", t.TempDir())
+	// Increase the log level; otherwise, its difficult to debug failures
+	viper.Set("Logging.Level", "Debug")
+	config.InitConfig()
+	err := config.InitServer()
+	require.NoError(t, err)
+
+	err = config.GeneratePrivateKey(param.Server_TLSKey.GetString(), elliptic.P256())
+	require.NoError(t, err)
+	err = config.GenerateCert()
+	require.NoError(t, err)
+
+	err = CheckXrootdEnv(true)
+	require.NoError(t, err)
+
+	err = SetUpMonitoring()
+	require.NoError(t, err)
+
+	configPath, err := ConfigXrootd(true)
+	require.NoError(t, err)
+
+	launchers, err := ConfigureLaunchers(false, configPath)
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		_ = daemon.LaunchDaemons(ctx, launchers)
```

Comment:
There is a failure when launching the daemons, it fails due to the lack of permission to use the key file: `XrdTLS: TLS_Context: Unable to use key file /tmp/TestOrigin1137965101/002/certificates/tls.key; permission denied`, which seems odd to me because that's a tmp directory and I'm pretty sure we add xrootd to the users/permissions when we do so.